test/functional/kernel/nested-loop/test-kernel-nested-loop.cpp.in

```diff
@@ -0,0 +1,154 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-21, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA_test-base.hpp"
+#include "RAJA_test-camp.hpp"
+
+#include "RAJA_test-forall-data.hpp"
+#include "RAJA_test-kernel-nested-loop-types.hpp"
+
+#include "test-kernel-nested-loop-@NESTED_LOOP_TYPE@.hpp"
+
+// ----------------------------------------------------------------------------
+// Sequential
+// ----------------------------------------------------------------------------
+using SequentialKernelNestedLoopExecPols = camp::list<
+
+    // Depth 2 Exec Pols
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::seq_exec  >,
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::loop_exec >,
+    NestedLoopData<DEPTH_2, RAJA::loop_exec, RAJA::seq_exec  >,
+    NestedLoopData<DEPTH_2, RAJA::loop_exec, RAJA::loop_exec >,
+
+    // Depth 3 Exec Pols
+    NestedLoopData<DEPTH_3, RAJA::seq_exec,  RAJA::seq_exec, RAJA::seq_exec >
+  >;
+
+// ----------------------------------------------------------------------------
+// OpenMP
+// ----------------------------------------------------------------------------
+#if defined(RAJA_ENABLE_OPENMP)
+
+using OpenMPKernelNestedLoopExecPols = camp::list<
+
+    // Depth 2 Exec Pols
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::omp_for_exec>,
+    NestedLoopData<DEPTH_2, RAJA::seq_exec,  RAJA::omp_parallel_for_exec >,
+    NestedLoopData<DEPTH_2, RAJA::omp_parallel_for_exec, RAJA::seq_exec >,
+    NestedLoopData<DEPTH_2, RAJA::omp_parallel_for_exec, RAJA::omp_parallel_for_exec >,
```

Comment:
Using nested OpenMP threading policies like this is problematic and may not yield correct behavior.  It may do something, but I don't think it will do what you think. I believe most compilers turn off nested parallel regions by default and you would have to set the environment variable OMP_NESTED (or whatever it is) to turn it on. @trws should weigh in on this.

Reply:
It's technically valid, but depending on the implementation and the nesting settings either means "give me all of my threads, distribute work across them, then add an unused parallel region and workshare across one thread" or "give me all of my threads, distribute work across them, have every thread create even more threads, then distribute work across all of those".  I'm not sure if any implementations do the latter anymore, but it used to be a problem with GCC and some others where they'd thread-bomb you with code like that.  I'd probably just skip it, honestly.  Having one in there that uses parallel for and inner simd might make sense though, or a parallel wrapped around a sequential outer and omp_for inner, depending on whether that's valid for this case.

Reply:
Thanks @trws. Your recommendation is what I was thing and discussing with Mike in chat about just now.

Reply:
Thanks for explaining further, I'll go ahead and alter the tests as suggested

Reply:
@mdavis36 also the gcc travis build is failing on ambiguous std::abs usage. You may have other ideas, but you could add a raja_test_abs<T>() method in a file in the test include directory and specialize it for float and integer types; i.e., call fabs for float and abs for int.

Reply:
Interesting, I'll go ahead and add a raja_test_abs utility to fix that, I can imagine that could be useful in the future.